### PR TITLE
Graceful highlight

### PR DESCRIPTION
**Pull request by hatched**
## Done

A user reported that the search and filter component was causing the Juju Dashboard to crash because it was receiving invalid data. This makes the source of the problem, the `highlightSubString` function more resilient to bad data in an attempt to solve the issue.

### QA steps

- Use the search and filter UI, it should continue to work as normal

## Fixes

Fixes: #477

### CODE REVIEW

File: src/components/SearchAndFilter/FilterPanelSection/FilterPanelSection.test.js [hunk 1]
```diff
@@ -10,12 +10,16 @@ const sampleData = {
 
 describe("Filter panel section", () => {
   it("renders", () => {
-    const wrapper = shallow(<FilterPanelSection data={sampleData} />);
+    const wrapper = shallow(
+      <FilterPanelSection data={sampleData} searchTerm="" />
```
Comment (anthonydillon) [inline src/components/SearchAndFilter/FilterPanelSection/FilterPanelSection.test.js:14]: Why is `searchTerm` a required prop here? Can it not be set to a default value of an empty string?

**Review by anthonydillon — CHANGES_REQUESTED** [review]
Question about the added prop to test.

File: src/components/SearchAndFilter/FilterPanelSection/FilterPanelSection.test.js (hunk 1 above)
Comment (hatched) [reply src/components/SearchAndFilter/FilterPanelSection/FilterPanelSection.test.js]: Updated :)

**Review by anthonydillon — APPROVED** [review]
LGTM :+1:

**Pull request merged**